### ISSUE TITLE
Enable users to save an opportunity and view/delete their saved opportunities

### ISSUE THREAD

**Issue by mxk0** [HHS/simpler-grants-gov#3118]
### Summary 

Today, when a user is interested in an opportunity, they have to bookmark the URL in their browser, save it to their to-do list, or use some other method of tracking leads outside of Simpler.Grants.gov. This adds overhead to the process of finding, keeping track of, and applying for an opportunity.

By allowing users to save opportunities on Simpler.Grants.gov, we can save them time and help minimize relevant opportunities slipping through the cracks.

### Acceptance criteria

- [ ] Users can save an opportunity
- [ ] Users can view their saved opportunities
- [ ] Users can open a saved opportunity
- [ ] Users can delete a saved opportunity

### Assumptions

- **NOTE: CREATING A BOOKMARK CAN AND SHOULD SHIP BEFORE WE HAVE E.G. DELETE BOOKMARK FUNCTIONALITY**
- Users would prefer authenticated bookmarks to browser bookmarks or other methods of compiling lists of opportunities
- Saved bookmarks for a single account (not shared amongst a team) are useful for a team member’s workflow. 

### Dependencies

- This feature depends on #2640
- This feature also depends on the CDN working for logged-in users 

### Not in scope for first iteration

- Email notifications about updates to saved opportunities. This will happen in a separate deliverable later in Quad 2.
- We will likely explore team-level saved opportunities at some point, but that is not in scope for now.
- Enabling users to filter or sort their saved opportunities.

**Comment by mxk0** [comment]
Notes from Refinement: 

- Design and frontend implementation for a user page
- Design for saving an opportunity (for both logged-out and logged-in users)
- Frontend implementation for saving an opportunity
- ~10 API tickets

**Comment by snyder-ux** [comment]
**UX & UI notes**
- "save" on all opportunities - regardless of grant status? ie does archive need a "save" button?
 - anonymous: click "save", heads up for need to sign in, sign in, returns to same opportunity with "saved!" message - details TBD
- authenticated: customized page with all saved opps.
   - zero state (maybe include a what/how to message)
   - list of opps. saved state
   - can delete and manage this list
   - if we have a "how to" msg for onboarding, what do we show after they've used it and return to a zero state?
